src/main/java/io/stargate/sgv2/jsonapi/util/naming/SchemaObjectNamingRule.java

```diff
@@ -0,0 +1,20 @@
+package io.stargate.sgv2.jsonapi.util.naming;
+
+import java.util.regex.Pattern;
+
+public abstract class SchemaObjectNamingRule implements NamingRule {
+
+  public static final int NAME_LENGTH = 48;
+  public static final Pattern PATTERN_WORD_CHARS = Pattern.compile("\\w+");
+
+  @Override
+  public abstract String name();
+
+  @Override
+  public boolean apply(String name) {
+    return name != null
+        && !name.isBlank()
```

Comment:
Can use `!name.isEmpty()` instead, regexp can handle "all whitespace" case (blank checks for empty (0 length) as well as all white space)